### ISSUE TITLE
Include system dependencies in datamap endpoint

### ISSUE THREAD

**Issue by SteveDMurphy** [ethyca/fides#987]
### Is your feature request related to a specific problem?

I want to be able to map system dependencies found across my data map

### Describe the solution you'd like

Have the datamap provide the appropriate fides keys and system dependencies as part of the datamap endpoint output

### Describe alternatives you've considered, if any

Just provide the fides keys and require a separate API request to map the system dependencies

### Additional context

This was originally requested by @LKCSmith - please update or modify to make sure the intent is covered and thank you!

**Comment by LKCSmith** [comment]
Endpoint: `GET /api/v1/datamap/${organization_fides_key}`

Problem: In order to set up and use the filters on the spatial datamap (if they are expected to use the same filters as the table), there needs to be a way to join the system data to the datamap data. However, the datamap response doesn't seem to have a `fides_key` .
We use system query because that is the only way to grab system dependencies (it doesn't appear in the datamap response), but if the datamap response could also give system dependencies, then we could just use that query instead of calling a system query.
Or if the datamap response could give a fides key to the system, then we could connect things in that way.

This blocks [this issue](https://github.com/ethyca/fidesctl-plus/issues/34).

**Comment by ssangervasi** [comment]
@LKCSmith 
>  In order to set up and use the filters on the spatial datamap (if they are expected to use the same filters as the table), there needs to be a way to join the system data to the datamap data.

[Follow up](https://github.com/ethyca/fidesctl-plus/issues/34#issuecomment-1216018201), I think you can go ahead with that ticket by having the systems filters only apply to the spatial view. We _might_ decide we want the filters to interact with each other, but at the moment that would probably be confusing.

However, your explanation totally applies to my [ticket (35)](https://github.com/ethyca/fidesctl-plus/issues/35) where clicking on a system is supposed to filter the table (I got sick before I noticed the blocker). I can fake it by matching on system name to start, but the keys are definitely what we want.

**Comment by LKCSmith** [comment]
@ssangervasi - Thank you for your input! I think I see what you're saying, and I'm glad that this change will also unblock you. I think the interactivity between the table and the spatial map is a separate issue from the one I am working on - my current issue is that I can't filter the spatial map at all (on its own) and would need some BE help as described above to get those filters for the spatial map going. I described this issue a little bit more in Slack...but I think either way we need this change. We should schedule a chat together to sync-up on the datamap since we're both working on it and have both been kind of off-and-on with being sick 😆 I have a feeling you might have some more information here that could be useful to me as well. I'll put something on your calendar, but if you are not feeling well, please feel free to re-schedule! I'm pretty open. 🙂
